touchforms/formplayer/experiments.py

```diff
@@ -59,42 +59,43 @@ def compare_list(control, candidate):
     return is_equal
 
 
-def compare_dict(control, candidate):
+def compare_dict(control, candidate, current_key):
     is_equal = True
     if control is not None and candidate is None:
+        logging.info('Key %s has null candidate but real control %s' % (current_key, control))
```

Comment:
i don't think you'll ever hit this because you're checking if candidate is a dict before you call this function

Reply:
good call, thanks